### PR TITLE
[Done!] Tasty Portals!

### PR DESCRIPTION
**Pull request by Rod995**
#### Summary
SUMMARY: Features "Artifacts can consume Portals."

#### Purpose of change
Adds an artifact effect that allows an artifact to consume a portal for charges.

#### Additional context
Artifacts can absorb portals for charges now, I hope.
I just need to do a bit more testing, just in case.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -12615,6 +12615,18 @@ void game::process_artifact( item &it, player &p )
                             it.charges++;
                         }
                         break;
+                    // Portals are energetic enough to charge the item.
+                    // Tears in reality are consumed too, but can't charge it.
+                    case ARTC_PORTAL:
+                        for( const tripoint &dest : m.points_in_radius( p.pos(), 1 ) ) {
+                            if( g->m.tr_at( dest ).loadid == tr_portal ) {
```
Comment (BevapDin) [inline src/game.cpp]: Please access the same thing the same way for consistency. Here you use just `m` (and implicitly mean `this->m`), but in the line above you use `g->m`. Please remove the `g->` (the code is in a `game` member functions).

**Comment by Rod995** [comment]
travis rebuilt

**Comment by Rod995** [comment]
travis rebuild

**Comment by Rod995** [comment]
Oi @BevapDin! 
It seems the requested change seems to have removed the field-devouring feature.
```Generated a fuckton of artifacts and got 7 with the portal charge type (charge_type 6 in the file). Of those, 6 generated with no secondary demands, while I got a crackling pyramid that likes being out in the rain. So I used up the charge, debugged the weather to sunny, then spawned in a portal and some dimensional fatigue.

It does indeed work right, not draining either while the weather is sunny. So I set the weather to rain, and soon the portal collapsed as expected. However, the tear in reality is not being absorbed as expected, and this persisted even after grabbing one of the other artifacts. It seems the code isn't finding and eating fatigue like it used to.```

File: src/game.cpp [hunk 2]
```diff
@@ -12615,6 +12615,18 @@ void game::process_artifact( item &it, player &p )
                             it.charges++;
                         }
                         break;
+                    // Portals are energetic enough to charge the item.
+                    // Tears in reality are consumed too, but can't charge it.
+                    case ARTC_PORTAL:
+                        for( const tripoint &dest : m.points_in_radius( p.pos(), 1 ) ) {
+                            if( m.tr_at( dest ).loadid == tr_portal ) {
```
[the remark this answers is not in the document]
Comment (Rod995) [reply src/game.cpp]: How would I do such a thing?

[the remark this answers is not in the document]
Comment (ZhilkinSerg) [reply src/game.cpp]: You can `break` from loop after first portal is found and consumed.

[the remark this answers is not in the document]
Comment (Rod995) [reply src/game.cpp]: Hey you!
Break thingie is in.

**Pull request merged**